src/cache/jsongMerge.js

```diff
@@ -65,7 +66,7 @@ function merge(config, cache, message, depth, path, fromParent, fromKey) {
             var values = config.values;
             values.push({
                 path: cloneArray(requestedPath),
-                value: message.type ? message.value : message
```

Comment:
@sdesai you are correct.  I think the good news is that we remove that statement and only ever use the value that is handed to us.  This could potentially strip some valuable meta data on a node by using the `.value` of the handed back value.